packages/iov-ethereum/src/abi.ts

```
@@ -112,6 +117,35 @@ export class Abi {
     return data.slice(32, 32 + length);
   }
 
+  public static decodeSwapProcessState(data: Uint8Array): SwapProcessState {
+    const key = Abi.decodeUint256(data);
+    const map: { readonly [key: string]: SwapProcessState } = {
+      1: SwapProcessState.Open,
+      2: SwapProcessState.Claimed,
+      3: SwapProcessState.Aborted,
+    };
+    const state: SwapProcessState | undefined = map[key];
+
+    if (state === undefined) {
+      throw new Error("Invalid swap process state");
+    }
+    return state;
+  }
+
+  public static decodeEventSignature(data: Uint8Array): SwapContractEvent {
```

I'd add a length check to fail early in case of programming errors